packages/protocol/contracts/governance/Proposals.sol

```diff
@@ -0,0 +1,330 @@
+    int256 participation = toFixed(
+      proposal.votes.yes.add(proposal.votes.no).add(proposal.votes.abstain)
+    ).divide(totalWeightFixed);
+    int256 adjustedYesNoRatio = (
+      participation > criticalBaseline ? participation : criticalBaseline
+    ).subtract(abstainRatio);
+    int256 support = yesRatio.divide(adjustedYesNoRatio);
+    return support;
+  }
+
+  /**
+   * @notice Returns the stage of a dequeued proposal.
+   * @param proposal The proposal struct.
+   * @param stageDurations The durations of the dequeued proposal stages.
+   * @return The stage of the dequeued proposal.
+   */
+  function getDequeuedStage(
+    Proposal storage proposal,
+    StageDurations storage stageDurations
+  )
+    internal
+    view
+    returns (Stage)
+  {
+    uint256 stageStartTime = proposal.timestamp.add(stageDurations.approval).add(
+      stageDurations.referendum
+    ).add(stageDurations.execution);
+    // solhint-disable-next-line not-rely-on-time
+    if (now >= stageStartTime) {
+      return Stage.Expiration;
+    }
+    stageStartTime = stageStartTime.sub(stageDurations.execution);
+    // solhint-disable-next-line not-rely-on-time
+    if (now >= stageStartTime) {
+      return Stage.Execution;
+    }
+    stageStartTime = stageStartTime.sub(stageDurations.referendum);
+    // solhint-disable-next-line not-rely-on-time
+    if (now >= stageStartTime) {
+      return Stage.Referendum;
+    }
+    return Stage.Approval;
+  }
+
+  /**
+   * @notice Returns a specified transaction in a proposal.
+   * @param proposal The proposal struct.
+   * @param index The index of the specified transaction in the proposal's transaction list.
+   * @return The specified transaction.
+   */
+  function getTransaction(
+    Proposal storage proposal,
+    uint256 index
+  )
+    public
+    view
+    returns (uint256, address, bytes memory)
+  {
+    require(index < proposal.transactions.length);
+    Transaction storage transaction = proposal.transactions[index];
+    return (transaction.value, transaction.destination, transaction.data);
+  }
+
+  /**
+   * @notice Returns an unpacked proposal struct with its transaction count.
+   * @param proposal The proposal struct.
+   * @return The unpacked proposal with its transaction count.
+   */
+  function unpack(
+    Proposal storage proposal
+  )
+    internal
+    view
+    returns (address, uint256, uint256, uint256)
+  {
+    return (
+      proposal.proposer,
+      proposal.deposit,
+      proposal.timestamp,
+      proposal.transactions.length
+    );
+  }
+
+  /**
+   * @notice Returns the referendum vote totals for a proposal.
+   * @param proposal The proposal struct.
+   * @return The yes, no, and abstain vote totals.
+   */
+  function getVoteTotals(
+    Proposal storage proposal
+  )
+    internal
+    view
+    returns (uint256, uint256, uint256)
+  {
+    return (proposal.votes.yes, proposal.votes.no, proposal.votes.abstain);
```

Comment:
nit: Let's make the order of these return values consistent with the struct ordering